code/parse/sexp.cpp

```diff
@@ -10812,6 +10818,11 @@ void sexp_explosion_effect(int n)
 		emp_duration = eval_num(n);
 		n = CDR(n);
 	}
+	if (n != -1)
+	{
+		use_emp_time_for_capship_turrets = is_sexp_true(n);
```

Comment:
For some reason, `is_sexp_true()` returns an `int`; since `use_emp_time_for_capship_turrets` is a `bool`, this line is generating a performance warning in VS (upgraded to an error, causing the AppVeyor failure). Either `is_sexp_true()` needs to return a `bool` or, if that's not feasible for whatever reason, the call needs to be changed to something like `!!is_sexp_true(n)` to get rid of the warning.

Reply:
I'll open a new PR for changing `is_sexp_true` to bool, since this happens all over the place.

Is `!!` a new operator?  I'm not getting search results on it, but being special characters, that's not unexpected.

Reply:
`!!` is just the double application of `!`. Since "not not true" is still true it doesn't change the boolean value but it will convert any integer into a true/false value.

Reply:
It's just the same operator twice, isn't it?  The suggestion is to cast to bool explicitly by inverting it (does the cast of the int to its inverse Boolean), and then invert that again, to get the Boolean equivalent of the integer.  If I'm reading that correctly.

Is there no (bool) operator in C++ for explicit casts though?  I would think it would complain about using !! to perform that cast implicitly too.

Reply:
Gotcha.

For the record, there are 324 instances of !! in the codebase, and most if not all of them appear to be the more conventional usage.

Reply:
I got ninja'd.  FWIW, I google'd C++ Boolean cast and found the third link mentioned using !! in the subject.  But I also found a StackOverflow answer that has a compelling argument against not using !! like this, and writing something easier to read and less likely to be mistaken for a typo.

https://stackoverflow.com/questions/1310344/why-use-when-converting-int-to-bool

Basically, just compare it to 0 explicitly to get a Boolean result from the comparison.

`use_emp_time_for_capship_turrets = is_sexp_true(n) != 0;`

Reply:
Yeah, most of the uses of `is_sexp_true` do it that way.  But this will shortly become moot, as the PR to change to bool is nearly ready to go.

Reply:
Ok, the other PR is merged and I've rebased this one.